Assets/HoloToolkit/Boundary/Scripts/BoundaryManager.cs

```diff
@@ -0,0 +1,162 @@
+        private Bounds boundaryBounds = new Bounds();
+
+        private bool renderFloor = true;
+        public bool RenderFloor
+        {
+            get
+            {
+                return renderFloor;
+            }
+            set
+            {
+                if (renderFloor != value)
+                {
+                    renderFloor = value;
+                    SetFloorRendering();
+                }
+            }
+        }
+
+        private void SetFloorRendering()
+        {
+            if (floorQuadInstance != null)
+            {
+                floorQuadInstance.SetActive(renderFloor);
+            }
+        }
+
+        private bool renderBoundary = true;
+        public bool RenderBoundary
+        {
+            get
+            {
+                return renderBoundary;
+            }
+            set
+            {
+                if (renderBoundary != value)
+                {
+                    renderBoundary = value;
+                    SetBoundaryRendering();
+                }
+            }
+        }
+
+        private void SetBoundaryRendering()
+        {
+            // TODO: BUG: Unity: configured bool always returns false.
+            if (UnityEngine.Experimental.XR.Boundary.configured)
+            {
+                UnityEngine.Experimental.XR.Boundary.visible = renderBoundary;
+            }
+        }
+
+        void Start()
+        {
+            // Render boundary if configured.
+            SetBoundaryRendering();
+
+            // Render the floor based on if you are in editor or immersive device.
+            RenderFloorQuad();
+        }
+
+        private void RenderFloorQuad()
+        {
+            if (FloorQuad != null && HolographicSettings.IsDisplayOpaque)
+            {
+                floorQuadInstance = Instantiate(FloorQuad);
+                floorQuadInstance.transform.SetParent(gameObject.transform.parent);
+
+#if UNITY_EDITOR
+                // So the floor quad does not occlude in editor testing, draw it lower.
+                floorQuadInstance.transform.localPosition = new Vector3(0, -3, 0);
+#else
+                // Inside immersive headset draw floor quad at Y value of dimensions.
+                Vector3 dimensions;
+                // TODO: BUG: Unity: configured bool always returns false.
+                // TODO: BUG: Unity: TryGetDimensions does not return true either.
+                //if (UnityEngine.Experimental.XR.Boundary.configured &&
+                //    UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+
+                if (UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+                {
+                    if (dimensions != null)
+                    {
+                        Debug.Log("Drawing floor at dimensions Y.");
+                        // Draw the floor at boundry Y.                    
+                        floorQuadInstance.transform.localPosition = new Vector3(0, dimensions.y, 0);
```

Comment:
There's also this which I'm not sure what to do with:

public enum Type
        {
            PlayArea = 0,
            TrackedArea = 1
        }